packages/chaire-lib-backend/src/utils/processManagers/OSRMProcessManager.ts

```diff
@@ -4,29 +4,27 @@
  * This file is licensed under the MIT License.
  * License text available at https://opensource.org/licenses/MIT
  */
-import Preferences from 'chaire-lib-common/lib/config/Preferences';
 import { getOsrmDirectoryPathForMode } from './OSRMServicePath';
 import ProcessManager from './ProcessManager';
 import { RoutingMode, routingModes } from 'chaire-lib-common/lib/config/routingModes';
 import osrmService from '../osrm/OSRMService';
 import OSRMMode from '../osrm/OSRMMode';
+import ServerConfig from '../../config/ServerConfig';
+import { setProjectConfiguration } from '../../config/server.config';
+import { _isBlank } from 'chaire-lib-common/lib/utils/LodashExtensions';
 
 const getServiceName = function (mode: RoutingMode = 'walking', port: number | null = 5000) {
     return `osrmMode__${mode}__port${port}`;
 };
 
 const routingModeIsAvailable = async function (routingMode: RoutingMode): Promise<boolean> {
-    const osrmServerPrefs = Preferences.get('osrmRouting', {});
-    if (
-        osrmServerPrefs.modes[routingMode] &&
-        osrmServerPrefs.modes[routingMode].enabled === true &&
-        osrmServerPrefs.modes[routingMode].autoStart !== true
-    ) {
+    const osrmModeConfig = ServerConfig.getRoutingEngineConfigForMode(routingMode, 'osrmRouting');
```

Comment:
J'ai oublié de le mentionner au premier review, mais il faudrait que tout reste compatible avec le fichier de config actuel (par exemple, la configuration de nos instances ne sera probablement pas mise à jour exactement en même temps que le prochain update de transition une fois cette PR mergée). Regarde ce qui a été fait pour la config `trRoutingCacheAllScenarios` dans `server.config.ts`, in the `setProjectConfiguration` function, if the old way of configuring is set, but now the new way, it is merged to make sure old configs still work. You can do something similar for the osrm routing's default preferences

Reply:
I added it to the setProjectConfiguration. I didn't push anything related to that to the tests since I didn't want to add deprecated stuff to any config file but I did test it locally and it works.